src/main/java/net/coderbot/iris/mixin/vertices/MixinBufferBuilder.java

```diff
@@ -76,8 +76,8 @@
 	@Inject(method = "begin", at = @At("RETURN"))
 	private void iris$afterBegin(int drawMode, VertexFormat format, CallbackInfo ci) {
 		if (extending) {
-			this.format = IrisVertexFormats.TERRAIN;
-			this.currentElement = IrisVertexFormats.TERRAIN.getElements().get(0);
+			this.format = format == DefaultVertexFormat.NEW_ENTITY ? IrisVertexFormats.ENTITY : IrisVertexFormats.TERRAIN;
```

Comment:
If the format here is IrisVertexFormats.ENTITY this will give terrain, which isn't correct. You'll need more robust logic here